src/Sentry/Protocol/Span.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    // https://develop.sentry.dev/sdk/event-payloads/span
+    public class Span : ISpan, IJsonSerializable
+    {
+        public SentryId SpanId { get; }
```

Comment:
The protocol really has all of these members being mutable?

Reply:
The protocol doesn't say anything about mutability. It doesn't even cover some of the fields at all.

Reply:
We should address that. Have a discussion and write it on the docs.
Ideally things would be immutable unless they need not to be.